corehq/apps/es/tests/test_case_search_es.py

```diff
@@ -212,3 +232,146 @@ def test_blacklisted_owner_ids(self):
+        with patch('corehq.pillows.case_search.domains_needing_search_index',
+                   MagicMock(return_value=[domain])):
+            CaseSearchReindexerFactory(domain=domain).build().reindex()
+
+    def _assert_query_runs_correctly(self, domain, input_cases, query, output):
+        for case in input_cases:
+            self._make_case(domain, case)
+        self._bootstrap_cases_in_es_for_domain(domain)
+        self.elasticsearch.indices.refresh(CASE_SEARCH_INDEX)
+        self.assertItemsEqual(
+            query.get_ids(),
+            output
+        )
+
+    def test_simple_case_property_query(self):
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c1', 'name': 'redbeard'},
+                {'_id': 'c2', 'name': 'blackbeard'},
+            ],
+            CaseSearchES().domain(self.domain).case_property_query("name", "redbeard"),
+            ['c1']
+        )
+
+    def test_multiple_case_search_queries(self):
+        query = (CaseSearchES().domain(self.domain)
+                 .case_property_query("name", "redbeard")
+                 .case_property_query("parrot_name", "polly"))
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c1', 'name': 'redbeard', 'parrot_name': 'polly'},
+                {'_id': 'c2', 'name': 'blackbeard', 'parrot_name': 'polly'},
+                {'_id': 'c3', 'name': 'redbeard', 'parrot_name': 'molly'}
+            ],
+            query,
+            ['c1']
+        )
+
+    def test_multiple_case_search_queries_should_clause(self):
+        query = (CaseSearchES().domain(self.domain)
+                 .case_property_query("name", "redbeard")
+                 .case_property_query("parrot_name", "polly", clause="should"))
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c1', 'name': 'redbeard', 'parrot_name': 'polly'},
+                {'_id': 'c2', 'name': 'blackbeard', 'parrot_name': 'polly'},
+                {'_id': 'c3', 'name': 'redbeard', 'parrot_name': 'molly'}
+            ],
+            query,
+            ['c1', 'c3']
+        )
+
+    def test_blacklisted_owner_ids(self):
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c1', 'owner_id': '123'},
+                {'_id': 'c2', 'owner_id': '234'},
+            ],
+            CaseSearchES().domain(self.domain).blacklist_owner_id('123'),
+            ['c2']
+        )
+
+    def test_missing_case_property(self):
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c2', 'name': 'blackbeard'},
+                {'_id': 'c3', 'name': ''},
+                {'_id': 'c4'},
+            ],
+            CaseSearchES().domain(self.domain).filter(case_property_missing('name')),
+            ['c3'] # todo; flag farid
```

Comment:
Hey @proteusvacuum we expect c4 to be here per the definition of `case_property_missing` but looks like it's not so. I am not going to fix it lest the existing uses break, but I did add a test and can add a caveat comment. Let me know if this is not how it's supposed to be.